core/helpers/EEH_Sideloader.helper.php

```diff
@@ -210,91 +213,131 @@ public function get_new_file_name()
+
+            // that's it.  let's allow for actions after file uploaded.
+            do_action('AHEE__EE_Sideloader__sideload_after', $this, $path);
+            return true;
+        } catch(Exception $exception) {
+            EE_Error::add_error($exception->getMessage(), __FILE__, __FUNCTION__, __LINE__);
+            return false;
+        }
+    }
 
-        // now we have the file, let's get it in the right directory with the right name.
-        $path = apply_filters('FHEE__EEH_Sideloader__sideload__new_path', $this->_upload_to . $this->_new_file_name, $this);
 
-        // move file in
-        if (false === @ rename($file, $path)) {
-            unlink($temp_file);
-            EE_Error::add_error(
+    /**
+     * returns TRUE if there IS an error, FALSE if there is NO ERROR
+     *
+     * @param array|WP_Error $response
+     * @return bool
+     * @throws RuntimeException
+     */
+    private function isResponseError($response)
+    {
+        if (! is_wp_error($response)) {
+            return false;
+        }
+        if (defined('WP_DEBUG') && WP_DEBUG) {
+            throw new RuntimeException(
                 sprintf(
-                    esc_html__('Unable to move the file to new location (possible permissions errors). This is the path the class attempted to move the file to: %s', 'event_espresso'),
-                    $path
-                ),
-                __FILE__,
-                __FUNCTION__,
-                __LINE__
+                    esc_html__(
+                        'The following error occurred while attempting to download the file from "%1$s":',
+                        'event_espresso'
+                    ),
+                    $this->_download_from,
+                    $response->get_error_message()
+                )
             );
-            return false;
         }
+        return true;
+    }
 
-        // set permissions
-        $permissions = apply_filters('FHEE__EEH_Sideloader__sideload__permissions_applied', $this->_permissions, $this);
-        chmod($path, $permissions);
-
-        // that's it.  let's allow for actions after file uploaded.
-        do_action('AHEE__EE_Sideloader__sideload_after', $this, $path);
 
-        // unlink tempfile
-        @unlink($temp_file);
+    /**
+     * returns TRUE if there IS an error, FALSE if there is NO ERROR
+     *
+     * @param array  $response
+     * @return bool
+     * @throws RuntimeException
+     */
+    private function isDownloadError(array $response)
+    {
+        $response_code = wp_remote_retrieve_response_code($response);
+        if ($response_code === 200) {
+            return false;
+        }
+        if (defined('WP_DEBUG') && WP_DEBUG) {
+            $msg = $response_code === 404
+                ? esc_html__(
+                    'Attempted to download a file from "%1$s" but encountered a "404 File Not Found" error.',
+                    'event_espresso'
+                )
```

Comment:
decided to add this new error, since I was getting the error below and surprisingly EE has not been translated into "Canadian", but the wording below made it seem worse than it was.